### DISCUSSION TITLE
Can one use `VisualElement.X` and `VisualElement.Y` to compute a visual element coordinates in its enclosing window?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/maui#18683] [Q&A] [answered]
I have created a demo project here: https://github.com/MartyIX/VisualElementXY202311 (.NET 8 RC2)

It [contains](https://github.com/MartyIX/VisualElementXY202311/blob/29c23563b58336ccfe83d6a8353fa7e3c2223885/MainPage.xaml.cs#L27) the method (added in the [commit](https://github.com/MartyIX/VisualElementXY202311/commit/29c23563b58336ccfe83d6a8353fa7e3c2223885)):

```cs
private (double x, double y) GetCoordinatesWithinWindow(VisualElement element)
{
    double x = 0;
    double y = 0;
    VisualElement? workingElement = element;
    while (workingElement is VisualElement visualElement)
    {
        x += visualElement.X;
        y += visualElement.Y;
        workingElement = workingElement.Parent as VisualElement;
    }

    return (x, y);
}
```

And I wonder whether this compution is fundamentally valid or not. I mean whether the computation gives what I claim it to be - ie coordinates of the element relative to the enclosing window.

Does anybody know?

btw: We experimented with this code and for an element we expected the X coordinate to be equal to `600`. However, it turned out to be `600.02489894573` or something. Hence the question about the validition of the approach.

**Comment by MartyIX** [comment]
@mattleibow @hartez Would you possibly know please?

**Reply by hartez** [reply, marked as answer]
This looks valid to me. The slight difference in your actual and expected values might be coming from rounding errors and conversions between native screen locations and the virtual locations (due to differing screen densities).
